metrics/src/metrics.rs

```diff
@@ -88,25 +88,34 @@ impl Default for MetricsAgent {
         Self::new(
             Arc::new(InfluxDbMetricsWriter::new()),
             Duration::from_secs(10),
+            //max per-second datapoint submission limit
+            4000,
         )
     }
 }
 
 impl MetricsAgent {
-    fn new(writer: Arc<MetricsWriter + Send + Sync>, write_frequency: Duration) -> Self {
+    fn new(
+        writer: Arc<MetricsWriter + Send + Sync>,
+        write_frequency: Duration,
+        rate_limit: usize,
+    ) -> Self {
         let (sender, receiver) = channel::<MetricsCommand>();
-        thread::spawn(move || Self::run(&receiver, &writer, write_frequency));
+        thread::spawn(move || Self::run(&receiver, &writer, write_frequency, rate_limit));
         MetricsAgent { sender }
     }
 
     fn run(
         receiver: &Receiver<MetricsCommand>,
         writer: &Arc<MetricsWriter + Send + Sync>,
         write_frequency: Duration,
+        rate_limit: usize,
     ) {
         trace!("run: enter");
         let mut last_write_time = Instant::now();
         let mut points = Vec::new();
+        let max_points: usize =
+            ((write_frequency.as_millis() as f32 * 1e-3) * rate_limit as f32) as usize;
```

Comment:
We don't really care about sub-second resolution, so this could be simplified to
```rust
let max_points = write_frequency.as_secs() as usize * max_points_per_sec;
```

Reply:
That's how I originally had it, but if we ever specify a write_frequency less than one second, the .as_secs() method just resolves to 0, which will cause weird behavior.  There is a test in this module that specifies write_frequency as Duration::from_millis(100), so I assume this behavior is wanted somewhere in the codebase.

Reply:
Na, 1 second resolution is plenty.   No need to submit to Influx faster than that ever.  We could even change `write_frequency: Duration` to `write_frequency_secs: usize` to make that more clear.